cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+    * Get container image hash.
+    * @param imageName Image name.
+    * @return A hash.
+    */
+  override def computeContainerImageHash(imageName: String): Map[String, String] = ???
+
+  /**
+    * Gather Docker image name from runtime attributes. It it's not present returns none.
+    * @param runtimeAttributes Runtime requirements for the task.
+    */
+  private def getDockerImage(runtimeAttributes: Map[String, String]): Option[String] = {
+    if (runtimeAttributes.filter(p => p._1.equals(DockerFlag)).size == 1 &&
+      !runtimeAttributes.filter(p => p._1.equals(DockerFlag)).values.head.isEmpty) {
+      Some(runtimeAttributes.filter(p => p._1.equals(DockerFlag)).values.head)
+    } else {
+      None
+    }
+  }
+
+  /**
+    * Extracts folder path from specific input file.
+    * @param file Absolute path from input file.
+    * @return File's folder.
+    */
+  private def extractFolder(file: String): Option[String] = {
+    try {
+      Some(file.substring(0, file.lastIndexOf("/")))
+    } catch {
+      case oooe: StringIndexOutOfBoundsException =>
+        logger.warn("Input with no valid folder pattern. It may be a intermediate value.", oooe)
+        None
+      case ex: Exception =>
+        logger.error("Unhandled exception.", ex)
+        throw ex
+    }
+  }
+
+  /**
+    * Creates docker command in order to execute the task into a container.
+    * @param image Docker image name.
+    * @return Command to execute.
+    */
+  private def buildDockerRunCommand(image: String): String = {
+    val dockerVolume = "-v %s:%s"
+    val inputFolder = task.inputs.filter(p => p._2.isInstanceOf[WdlFile]).map(p => extractFolder(p._2.toWdlString.replace("\"", "")))
+      .filter(p => p.isDefined).map(p => p.get).toList.distinct
+    val inputVolumes = inputFolder match {
+      case a: List[String] => inputFolder.map(v => dockerVolume.format(v, v)).mkString(" ")
+      case _ => ""
+    }
+    val outputVolume = dockerVolume.format(executionDir, executionDir)
+    log.debug(s"DockerInputVolume: $inputVolumes")
+    log.debug(s"DockerOutputVolume: $outputVolume")
+    val dockerCmd = s"docker run %s %s --rm %s %s" //TODO: make it configurable from file.
+    dockerCmd.format(inputVolumes, outputVolume, image, argv.mkString(" ")).replace("  ", " ")
+  }
+
+  /**
+    * Writes the script file containing the user's command from the WDL as well
+    * as some extra shell code for monitoring jobs
+    */
+  private def writeBashScript(taskCommand: String, executionPath: Path): Unit = {
+    script.write(
+      s"""#!/bin/sh
+          |cd $executionPath
+          |$taskCommand
+          |echo $$? > rc
+          |""".stripMargin)
+  }
+
+  /**
+    * Resolves absolute path for output files. If it is not a file, it will returns same value.
+    * @param output Pair of WdlType and WdlValue
+    * @return WdlValue with absolute path if it is a file.
+    */
+  private def resolveOutputValue(output: (WdlType, Try[WdlValue])): WdlValue = {
```

Comment:
Personally (and not everyone here agrees with me) I see no point in adding special monikers to point out a wrapping monad on a type (e.g. maybeFoo, futureFoo, etc) so it's just a type and value. 

As an aside, we traditionally have grossly overused Try instead of ErrorOr (a type alias from a package object) which is [String, A] instead of [Throwable, A] of Try. If the Try is coming from code already existing on our end so be it but if it's being newly created the ErrorOr is likely better.